command/console.go

```diff
@@ -128,24 +100,26 @@ func (*ConsoleCommand) AutocompleteFlags() complete.Flags {
 	}
 }
 
-func (c *ConsoleCommand) modePiped(session *REPLSession) int {
+func (c *ConsoleCommand) modePiped(cfg packer.Evaluator) int {
 	var lastResult string
 	scanner := bufio.NewScanner(wrappedstreams.Stdin())
+	ret := 0
 	for scanner.Scan() {
-		result, err := session.Handle(strings.TrimSpace(scanner.Text()))
-		if err != nil {
-			return 0
+		result, _, diags := cfg.EvaluateExpression(strings.TrimSpace(scanner.Text()))
+		if len(diags) > 1 {
```

Comment:
In the case that we only have one item in diags should we print it? Or should this be if `len(diags) > 0`?

Reply:
Yes totally !

Reply:
```suggestion
		if len(diags) > 0 {
```